src/app/games/go/GoRules.ts

```diff
@@ -200,16 +200,16 @@ export class GoRules extends Rules<GoMove, GoState, GoLegalityStatus> {
 
         const newBoard: GoPiece[][] = state.getCopiedBoard();
         const currentTurn: number = state.turn;
-        const currentPlayer: number = currentTurn%2 === 0 ? GoPiece.BLACK.value : GoPiece.WHITE.value;
+        const currentPlayer: GoPiece = currentTurn % 2 === 0 ? GoPiece.BLACK : GoPiece.WHITE;
```

Comment:
`GoPiece.ofPlayer(this.getCurrentPlayer())` plutôt ?

Reply:
Voilà, GoPiece est encore mieux maintenant! Check tu !